src/PSRule.Rules.Azure/rules/Azure.Resource.Rule.ps1

```diff
@@ -34,3 +34,48 @@ Rule 'Azure.ResourceGroup.Name' -Ref 'AZR-000168' -Type 'Microsoft.Resources/res
     # Can't end with period.
     $Assert.Match($PSRule, 'TargetName', '^[-\w\._\(\)]*[-\w_\(\)]$');
 }
+
+
+# Synopsis: Ensure all properties named used for setting a username within a deployment are expressions (e.g. an ARM function not a string)
+Rule 'Azure.Resource.adminUsername' -Ref 'AZR-000284' -Type 'Microsoft.Resources/deployments' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {  
+    $deploymentTemplate = @($TargetObject.properties.template);
+
+    if (($deploymentTemplate.resources).Length -eq 0 ) {
+        return $Assert.Pass();
+    } else {
+        foreach ($resource in $deploymentTemplate.resources.properties) {
+            global:FindAdminUsername -InputObject $resource
+        }
+    }
+}
+
+## Functions
+function global:FindAdminUsername {
+    param ([PSObject]$InputObject)
+    process {
+    foreach ($property in $InputObject.PSObject.properties) {
+        ## Loop again if the property is a nested object
+        if($property.Value.GetType().Name -eq 'PSCustomObject') {
+            foreach($nestedProperty in $property.Value.PSObject.Properties ){
+                global:CheckSensitiveProperties -InputObject $nestedProperty
+            }
+        } elseif ($property.value.GetType().Name -eq 'String') {
+            global:CheckSensitiveProperties -InputObject $property
+        } 
+      }
+    }
+  }
+
+  function global:CheckSensitiveProperties {
+    param (
+        [PSObject]$InputObject
+        )
+    process {
+        if ($Configuration.GetStringValues('AZURE_TEMPLATE_SENSITIVE_PROPERTY_NAMES') -contains $InputObject.Name) {
+            $cleanValue = [PSRule.Rules.Azure.Runtime.Helper]::CompressExpression($InputObject.Value);
+            $Assert.Match($cleanValue, '.', '\[[^\]]+\]')   
```

Comment:
Thinking on this more and with reference to docs, we should probably only allow from parameters instead of also a hardcoded value from a variable.